dmoj/executors/base_executor.py

```diff
@@ -59,45 +59,45 @@ def cleanup(self):
     def __del__(self):
         self.cleanup()
 
-    def _file(self, *paths):
+    def _file(self, *paths: str) -> str:
         # Defer creation of temporary submission directory until first file is created,
         # because we may not need one (e.g. for cached executors).
         if self._dir is None:
             self._dir = tempfile.mkdtemp(dir=self._tempdir)
         return os.path.join(self._dir, *paths)
 
     @classmethod
-    def get_executor_name(cls):
+    def get_executor_name(cls) -> list:
         return cls.__module__.split('.')[-1]
 
-    def get_executable(self):
+    def get_executable(self) -> None:
```

Comment:
This is `Union[str, None]`.

Reply:
This function returns `None` on line 74. Would you please show me where it has the option to return a `str` before I make this change?

Reply:
`BaseExecutor` is a sort of abstract class for `CompiledExecutor`, `ScriptExecutor` etc. which actually implement these functions.